projects/opendr_ws/src/perception/scripts/rgbd_hand_gesture_recognition.py

```diff
@@ -0,0 +1,166 @@
+        depth_sub = message_filters.Subscriber(input_depth_image_topic, ROS_Image)
+        #synchronize image and depth data topics
+        ts = message_filters.TimeSynchronizer([image_sub, depth_sub], 10)
+        ts.registerCallback(self.callback)
+        
+        self.bridge = ROSBridge()
+
+        # Initialize the gesture recognition
+        self.gesture_learner = RgbdHandGestureLearner(n_class=16, architecture="mobilenet_v2", device=device)
+        model_path = './mobilenet_v2'
+        if not os.path.exists(model_path):
+            self.gesture_learner.download(path=model_path)
+        self.gesture_learner.load(path=model_path)
+        
+        #mean and std for preprocessing, based on HANDS dataset
+        self.mean = np.asarray([0.485, 0.456, 0.406, 0.0303]).reshape(1, 1, 4)
+        self.std = np.asarray([0.229, 0.224, 0.225, 0.0353]).reshape(1, 1, 4) 
+        
+
+    def listen(self):
+        """
+        Start the node and begin processing input data
+        """
+        rospy.init_node('opendr_gesture_recognition', anonymous=True)
+        rospy.loginfo("RGBD gesture recognition node started!")
+        rospy.spin()
+
+    def callback(self, image_data, depth_data):
+        """
+        Callback that process the input data and publishes to the corresponding topics
+        :param image_data: input image message
+        :type image_data: sensor_msgs.msg.Image
+        :param depth_data: input depth image message
+        :type depth_data: sensor_msgs.msg.Image
+        """
+        
+        # Convert sensor_msgs.msg.Image into OpenDR Image and preprocess
+        image = self.bridge.from_ros_image(image_data, encoding='bgr8')
+        depth_data.encoding='mono16'
+        depth_image = from_ros_image_to_depth(depth_data, encoding='mono16')
+        img = self.preprocess(image, depth_image)
+        
+        # Run gesture recognition
+        gesture_class = self.gesture_learner.infer(img)
+
+        #  Publish results
+        ros_gesture = from_category(gesture_class)
+        self.gesture_publisher.publish(ros_gesture)
+        
+    def preprocess(self, image, depth_img):
+        '''
+        Preprocess image, depth_image and concatenate them
+        :param image_data: input image 
+        :type image_data: engine.data.Image
+        :param depth_data: input depth image 
+        :type depth_data: engine.data.Image
+        '''
+        image = image.numpy() / (2**8 - 1)
+        depth_img = depth_img.numpy() / (2**16 - 1)
+
+        # resize the images to 224x224
+        image = cv2.resize(image, (224, 224))
+        depth_img = cv2.resize(depth_img, (224, 224))
+
+        # concatenate and standardize
+        img = np.concatenate([image, np.expand_dims(depth_img, axis=-1)], axis = -1)
+        img = (img - self.mean) / self.std
+        
+        img = Image(img, dtype=np.float32)
+        return img
+
+
+def from_category(prediction, source_data=None):
```

Comment:
Please move this generic functionality to [projects/opendr_ws/src/ros_bridge/src/opendr_bridge/bridge.py](https://github.com/tasostefas/opendr_internal/blob/master/projects/opendr_ws/src/ros_bridge/src/opendr_bridge/bridge.py).

Reply:
Also, after moving the corresponding function to bridge.py please update the corresponding documentation (https://github.com/tasostefas/opendr_internal/blob/master/docs/reference/rosbridge.md).